plugins/events/create.js

```diff
@@ -219,9 +219,16 @@ module.exports = () => ({
                                         )
                                         // User has good permissions, create an event
                                         .then(() =>
-                                            scm.getCommitSha(scmConfig).then(sha => {
-                                                payload.sha = sha;
-                                            })
+                                            scm
+                                                .getCommitSha(scmConfig)
+                                                .then(sha => {
+                                                    payload.sha = sha;
+                                                })
+                                                .catch(err => {
+                                                    if (err.status === 404) {
+                                                        throw boom.notFound(err.message);
```

Comment:
```suggestion
                                                        throw boom.notFound(err);
```

to keep the information of the original error.